### ISSUE TITLE
Update Quasar Users Table to Handle Updated Records from Northstar

### ISSUE THREAD

**Issue by jamjensen** [DoSomething/quasar#373]
_TASK_
-------------------
#### Description
As a data engineer, I want to extend the functionality of the Northstar-to-Quasar user ingestion ETL script so that it ingests updated records (in a given backfill time-period) from Northstar; such as,`last_accessed`, `last_logged_in`, and user profile fields, e.g.`phone_number`.

#### Why This is Important
This is necessary for us to (1) analyze user behavior on site; (2) have the most up-to-date user records; (3) remove reliance on phoenix database for user data. 

#### Definition of Done
**Given** Northstar API endpoint is fully exposed
**When** Sena confirms that the extended ETL script pulls updated Northstar records based on a given backfill time-period
**Then** We (team Quasar) will perpetually ingest updated Northstar records into Quasar

**Comment by jamjensen** [comment]
@sheyd are you still blocked by #361?

**Comment by DFurnes** [comment]
@jamjensen That pull request was deployed to production earlier today! 🚀

**Comment by sheyd** [comment]
I'm not blocked by #361 anymore. I've updated the DS Northstar scraper code to call the updated NS API endpoints. I'm updating the ETL process to parse and update the DB.

**Comment by sheyd** [comment]
Got the DS Northstar Scraper updated to handle new endpoints. Finishing updated ETL code.

**Comment by jamjensen** [comment]
@sheyd were you able to finish updating Northstar-to-Quasar user ingestion ETL script?

**Comment by sheyd** [comment]
Code has been merged, and I've updated the Jenkins Northstar -> Users table ingestion to use the new backfill process. It pulls updated and newly created members based on an hourly backfill every 15 minutes (up from 5). Not the best setup as, there is overlap 3-4 times an hour, and it lowers our 5 minute updates, but I don't want to tax the Northstar API too much, and our 5 minute runs were getting a bit onerous. 

I've done a test run and have seen positive results in the DB. Assigning to @jamjensen for further confirmation throughout the day that we're seeing updated `last_accessed` and `last_logged_in` fields to confirm updates from backfill, as well as `northstar_created_at_timestamp` fields that are propagating from today's date(7/13/17), ensuring we have full data fidelity. 

From the code side and ETL side, this work is complete, unless erroneous data is coming through.

**Comment by sheyd** [comment]
So the backfill for 60 and 30 days has failed multiple times. The queries seem to be particularly expensive for Northstar, so I've taken a different tactic to get us back up to date. I've kicked off a full backfill migration again, and then will backfill for the 4 day time period. 

Tagging in @DFurnes. I'm not sure what code optimizations or indexes we might be able the add to the DB to help, but CPU spiking has been a tad insane. I'll put the Mongo Atlas metric link in Slack for you to peruse. I'm also happy to pour over some logs with you whenever you get around to looking at this, so we can do some testing again Thorstar™ before pushing Prod to the edge again.

**Comment by DFurnes** [comment]
@sheyd Can we look over the queries you're making sometime today? I'm testing some queries against production (loading records updated on 2/1/2017 hour-by-hour) and performance looks [pretty decent](https://user-images.githubusercontent.com/583202/28220358-22998a94-688d-11e7-9d90-ccb7332c0b19.png).

```
2017-02-01 0:00-1:00 - Loaded 9833 records in 9 seconds (100 requests).
2017-02-01 1:00-2:00 - Loaded 11238 records in 10 seconds (114 requests).
2017-02-01 2:00-3:00 - Loaded 10199 records in 10 seconds (103 requests).
2017-02-01 3:00-4:00 - Loaded 9482 records in 9 seconds (96 requests).
2017-02-01 4:00-5:00 - Loaded 11500 records in 11 seconds (116 requests).
2017-02-01 5:00-6:00 - Loaded 8641 records in 8 seconds (88 requests).
2017-02-01 6:00-7:00 - Loaded 9615 records in 10 seconds (98 requests).
2017-02-01 7:00-8:00 - Loaded 9699 records in 9 seconds (98 requests).
2017-02-01 8:00-9:00 - Loaded 8718 records in 9 seconds (89 requests).
2017-02-01 9:00-10:00 - Loaded 9753 records in 9 seconds (99 requests).
2017-02-01 10:00-11:00 - Loaded 8807 records in 11 seconds (90 requests).
2017-02-01 11:00-12:00 - Loaded 8654 records in 8 seconds (88 requests).
2017-02-01 12:00-13:00 - Loaded 9339 records in 10 seconds (95 requests).
2017-02-01 13:00-14:00 - Loaded 9141 records in 9 seconds (93 requests).
2017-02-01 14:00-15:00 - Loaded 10579 records in 11 seconds (107 requests).
2017-02-01 15:00-16:00 - Loaded 12049 records in 11 seconds (122 requests).
2017-02-01 16:00-17:00 - Loaded 10659 records in 11 seconds (108 requests).
2017-02-01 17:00-18:00 - Loaded 13424 records in 13 seconds (136 requests).
2017-02-01 18:00-19:00 - Loaded 17439 records in 18 seconds (176 requests).
2017-02-01 19:00-20:00 - Loaded 23979 records in 26 seconds (241 requests).
2017-02-01 20:00-21:00 - Loaded 26561 records in 30 seconds (267 requests).
2017-02-01 21:00-22:00 - Loaded 28691 records in 33 seconds (288 requests).
2017-02-01 22:00-23:00 - Loaded 29660 records in 33 seconds (298 requests).
2017-02-01 23:00-24:00 - Loaded 28591 records in 34 seconds (287 requests).
```

Here's an example URL I'm loading:

```
https://profile.dosomething.org/v1/users?page=97&pagination=cursor&after%5Bupdated_at%5D=2017-02-01T00:00:00%2B00:00&before%5Bupdated_at%5D=2017-02-01T01:00:00%2B00:00&limit=100
```

**Comment by sheyd** [comment]
@DFurnes Sure! Let's talk when you get a chance, or revisit Monday if you're 🔙 🚗 ⏲ today.

**Comment by sheyd** [comment]
Full Back fill is still going, but talked to Dave, and the real performance bottleneck here is how Mongo counts users when queried against a large date range. There are 3 things I have to do to make this ticket really stick and fully functional. @jamjensen It might be worth putting this into another ticket, something like "Performance Tuning for Northstar -> User Ingestion ETL Script":
- [ ] Update Northstar Web Scraper calls for updated_at/created_at filters to use the `pagination:cursor` field for fast filtering.
- [ ] Update full and diff backfills in ETL script to breakdown any time period into individual hour chunks to iterate over users for more efficient querying.
- [ ] Test the code.

**Comment by sheyd** [comment]
The Northstar full backfill finally completed, and I've ran the backfill diff for the 6 days or so since then successfully. We're also running hourly backfills the inefficient way every 15 minutes, which is getting us updated records. @jamjensen If you can provide some confirmation, the only steps left on this are the performance refactoring, which hopefully Xianny can get to next week if our Jenkins/code-refactor goes well tomorrow.

**Comment by sheyd** [comment]
Closing this since work is done in deference to #398.
